virttest/qemu_devices/qdevices.py

```diff
@@ -1616,7 +1618,110 @@ def disable(self, monitor):
+        :type aobject: str
+        :param child_bus: List of buses, which this device provides.
+        :type child_bus: QSparseBus
+        """
+        aid = '%s_%s' % (name, aobject)
+        super(QDaemonDev, self).__init__(name, aobject=aobject, child_bus=child_bus)
+        self.set_aid(aid)
+        self.set_param('name', name)
+        self._daemon_process = None
+
+    def is_daemon_alive(self):
+        """Check whether daemon is alive."""
+        return bool(self._daemon_process and self._daemon_process.is_alive())
+
+    def start_daemon(self):
+        """Start daemon."""
+        cmd = self.get_param('cmd')
+        name = self.get_param('name')
+        run_bg_kwargs = self.get_param('run_bg_kwargs', {})
+        status_active = self.get_param('status_active')
+        read_until_timeout = self.get_param('read_until_timeout', 5)
+        start_until_timeout = self.get_param('start_until_timeout', 1)
+
+        if cmd is None:
+            logging.warn('No provided command to start %s daemon.', name)
+            self._daemon_process = None
+
+        if self.is_daemon_alive():
+            return
+
+        logging.info('Running %s daemon command %s.', name, cmd)
+        self._daemon_process = aexpect.run_bg(cmd, **run_bg_kwargs)
+        if status_active:
+            self._daemon_process.read_until_any_line_matches(
+                    status_active, timeout=read_until_timeout)
+        else:
+            time.sleep(start_until_timeout)
+        logging.info("Created %s daemon process with parent PID %d.",
+                     name, self._daemon_process.get_pid())
+
+    def stop_daemon(self):
+        """Stop daemon."""
+        if self._daemon_process is not None:
+            try:
+                if not utils_misc.wait_for(lambda: not self.is_daemon_alive(),
+                                           self.get_param('stop_timeout', 3)):
+                    raise DeviceError('The %s daemon is still alive.' %
+                                      self.get_param('name'))
+            finally:
+                self._daemon_process.close()
+                self._daemon_process = None
+
+    def clear(self):
+        """Clear daemon."""
+        try:
+            self.stop_daemon()
+        except DeviceError:
+            pass
+
+    def hotplug(self, monitor):
+        """Hot plug daemon."""
+        self.start_daemon()
+
+    def verify_hotplug(self, out, monitor):
+        """Verify daemon after hotplug."""
+        return self.is_daemon_alive()
+
+    def unplug(self, monitor):
+        """Unplug daemon."""
+        self.stop_daemon()
+
+    def verify_unplug(self, out, monitor):
+        """Verify the status of daemon."""
+        return not self.is_daemon_alive()
+
+    def get_qid(self):
+        """Get qid."""
+        return self.get_aid()
+
+    def cmdline(self):
+        """Start daemon command line."""
+        self.start_daemon()
+        return ''
+
+    def __eq__(self, other, dynamic=True):
```

Comment:
The parameter `dynamic` seems useless.

Reply:
Hi @vivianQizhu 
Yes, seems the `dynamic` is useless.
like virtio-fs implemention
```
    def hotplug(self, monitor):
        self._start_daemon()

    def verify_hotplug(self, out, monitor):
        return self._daemon_alive()

    def unplug(self, monitor):
        self._stop_daemon()

    def verify_unplug(self, out, monitor):
        """Verify the status of virtiofs daemon."""
        return not self._daemon_alive()
```
also, the `monitor` is useless, but it may be called by somewhere in the future, 
for example:
```
def simple_hotplug(self, device, monitor, bus=None):
...
    out = device.hotplug(monitor)
```
And The monitor is useless since they inherit the base class which has defined the arguments.
here better to keep the arguments with the base class if we wanna override them, I think.

But I saw that the `QBaseDevice__eq__(self, dev2, dynamic=True)` the dynamic doesn't do anything.
So I don't know why keep it here, maybe other reasons?

Delete the argument `dynamic` here also look good to me.
How about your idea?
Thanks.

Reply:
> also, the `monitor` is useless, but it may be called by somewhere in the future,

This is definitely not a good idea. It might be a dead code and leads some one copy your wrong code some day as you copy the useless `dynamic`. Please remove them. Thanks.

Reply:
Hi @vivianQizhu 
All right, have updated. please help me to review, thanks.

Reply:
I think we should remove `dynamic=True` here.

Reply:
Hi @vivianQizhu
Have updated. please help me to review, thanks.